src/System.Net.HttpListener/tests/HttpListenerFactory.cs

```diff
@@ -7,15 +7,24 @@
 using System.Net.Sockets;
 using System.Runtime.InteropServices;
 using System.Text;
+using System.IO;
 
 namespace System.Net.Tests
 {
     // Utilities for generating URL prefixes for HttpListener
     public class HttpListenerFactory : IDisposable
     {
-        const int MinPort = 1025;
-        private static readonly object s_createListenerLock = new object();
-        private static int s_nextPort = MinPort;
+        private const int MaxStartAttempts = 500;
+        private static int s_minPort;
+        private static int s_maxPort;
+        private static readonly Random s_random = new Random();
```

Comment:
didn't notice these are tests - I do not know if there are any negative implications of that then